student/views/registration.py

```diff
@@ -71,8 +73,20 @@ def done(self, form_list, **kwargs):
         except IntegrityError:
             return redirect('basics_register')
 
+
 class PaymentPreviewView(View):
     def get(self, request, registration_id):
+        registration = Registration.objects.get(pk=registration_id)
+        payment_amount = settings.CODING_BASICS_REGISTRATION_FEE_SGD
+        if registration.referral_code:
+            coupon = Coupon.objects.get(code=registration.referral_code)
+            coupon_effect = coupon.effects.filter(couponable_type='Course', couponable_id=1).first()
```

Comment:
just a note to @rocketkai - we will change this to cater for list of discounts in future, probably when we handle tiered payments. will see how it goes.

Reply:
Noted, thank you!